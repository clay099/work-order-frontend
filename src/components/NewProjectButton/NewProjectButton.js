import React from "react";
import { Button, Grid } from "@material-ui/core";
import { useHistory } from "react-router-dom";
import { makeStyles } from "@material-ui/core/styles";

const NewProjectButton = ({ justify }) => {
	const history = useHistory();

	const useStyles = makeStyles((theme) => ({
		newProjectButton: {
			marginTop: theme.spacing(2),
		},
	}));

	const classes = useStyles();

	return (
		<Grid
			container
			spacing={2}
			justify={justify}
			className={classes.newProjectButton}
		>
			<Grid item>
				<Button
					variant="contained"
					color="primary"
					onClick={() => history.push("/user/newproject")}
				>
					Create New Project
				</Button>
			</Grid>
		</Grid>
	);
};

export default NewProjectButton;
