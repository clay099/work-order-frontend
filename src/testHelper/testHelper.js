// testHelper.js
import React from "react";
import { render as rtlRender } from "@testing-library/react";
import { createStore } from "redux";
import { Provider } from "react-redux";
// Import your own reducer
import reducer from "../reducers/rootReducer";

function render(
	ui,
	{
		initialState,
		store = createStore(reducer, initialState),
		...renderOptions
	} = {}
) {
	function Wrapper({ children }) {
		return <Provider store={store}>{children}</Provider>;
	}
	return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

const tokenState = { token: "token" };
const userTypeUserState = { userType: "user" };
const userTypeTradesmenState = { userType: "user" };
const idState = { id: 1234 };
const userDetailsState = { userDetails: "userDetails" };
const handleSubmitState = { handleSubmit: jest.fn() };
const handleChangeState = { handleChange: jest.fn() };
const disabledState = { disabled: false };
const disabledEmailState = { disabledEmail: false };
const buttonTextState = { buttonText: "buttonText" };
const formDataState = {
	formData: {
		id: 1,
		firstName: "firstName",
		lastName: "lastName",
		email: "email",
		password: "password",
		newPassword: "newPassword",
		confirmNewPassword: "confirmNewPassword",
	},
};

// re-export everything
export * from "@testing-library/react";
// override render method
export {
	render,
	tokenState,
	userTypeUserState,
	userTypeTradesmenState,
	idState,
	userDetailsState,
	handleSubmitState,
	handleChangeState,
	disabledState,
	disabledEmailState,
	buttonTextState,
	formDataState,
};
