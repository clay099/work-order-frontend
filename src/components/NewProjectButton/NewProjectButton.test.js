import React from "react";
import NewProjectButton from "./NewProjectButton";
import { render } from "../../testHelper/testHelper";
console.error = jest.fn();

const params = {
	justify: "center",
};

describe("NewProjectButton Component", () => {
	// smoke test
	test("renders without crashing", () => {
		render(<NewProjectButton {...params} />);
	});

	// snapshot test
	test("matches snapshot", () => {
		const { asFragment } = render(<NewProjectButton {...params} />);
		expect(asFragment).toMatchSnapshot();
	});

	test("value to be in the document", () => {
		const { getByText } = render(<NewProjectButton {...params} />);
		const value = getByText("Create New Project");
		expect(value).toBeInTheDocument();
	});
});
