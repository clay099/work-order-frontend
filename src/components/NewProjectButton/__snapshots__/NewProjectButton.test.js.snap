// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewProjectButton Component matches snapshot 1`] = `[Function]`;
